src/strat_engine/blockdev.rs

```diff
@@ -50,92 +50,112 @@ pub struct BlockDev {
+                },
+                mdab: MDA {
+                    last_updated: Timespec::new(0, 0),
+                    used: 0,
+                    length: (*mda_size / 2 * SECTOR_SIZE) as u32,
+                    crc: 0,
+                    offset: SectorOffset(*mda_size / 2),
+                },
+                mda_sectors: mda_size,
+                reserved_sectors: MDA_RESERVED_SIZE,
+            };
+
+            try!(bd.write_sigblock());
+            bds.push(bd);
+        }
+        Ok(bds)
+    }
 
-        if !force {
-            let mut buf = [0u8; 4096];
-            try!(f.read(&mut buf));
+    /// Gets device and device sizes, and returns an error if devices
+    /// cannot be used by Stratis.
+    fn dev_info(paths: &[&Path], force: bool) -> EngineResult<Vec<(Device, u64)>> {
+        let mut dev_infos = Vec::new();
+        for path in paths {
+            let metadata = try!(fs::metadata(path));
 
-            if buf.iter().any(|x| *x != 0) {
+            if !metadata.file_type().is_block_device() {
                 return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                          format!("First 4K of {} is not \
-                                                                   zeroed, need to use --force",
+                                                          format!("{} is not a block device",
                                                                   path.display()))));
             }
-        }
 
-        let dev_size = try!(blkdev_size(&f));
-        if dev_size < MIN_DEV_SIZE {
-            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} too small, {} minimum",
-                                                             path.display(),
-                                                             ByteSize::b(MIN_DEV_SIZE as usize)
-                                                                 .to_string(true)))));
-        }
-
-        let mut bd = BlockDev {
-            stratis_id: stratis_id.to_owned(),
-            id: Uuid::new_v4().to_simple_string(),
-            dev: dev,
-            path: path.to_owned(),
-            sectors: Sectors(dev_size / SECTOR_SIZE),
-            mdaa: MDA {
-                last_updated: Timespec::new(0, 0),
-                used: 0,
-                length: (*mda_size / 2 * SECTOR_SIZE) as u32,
-                crc: 0,
-                offset: SectorOffset(0),
-            },
-            mdab: MDA {
-                last_updated: Timespec::new(0, 0),
-                used: 0,
-                length: (*mda_size / 2 * SECTOR_SIZE) as u32,
-                crc: 0,
-                offset: SectorOffset(*mda_size / 2),
-            },
-            mda_sectors: mda_size,
-            reserved_sectors: MDA_RESERVED_SIZE,
-        };
+            let dev = try!(Device::from_str(&path.to_string_lossy()));
```

Comment:
Please try to move ```dev``` down to its first use, at around line 155. Or, even better, don't calculate it in this method, calculate it in the chunk of code starting at180.